File: Samples/FSharpDemo/MeshVisualization.fs
namespace FSharpDemo

open System.Diagnostics
open Autodesk.Revit.Attributes
open Autodesk.Revit.UI.Selection
open JetBrains.Annotations
open Nice3point.Revit.Toolkit.External
open RevitDevTool.FSharpDemo.Extensions.GeometryExtensions

[<Transaction(TransactionMode.Manual)>]
[<UsedImplicitly>]
type MeshVisualization() =
    inherit ExternalCommand()

    override this.Execute() =
        try
            let meshRef = this.Application.ActiveUIDocument.Selection.PickObject(ObjectType.Element, "Select Mesh Element")
            let meshes = getMeshes (this.Application.ActiveUIDocument.Document.GetElement(meshRef))
            Trace.Write(meshes |> List.head)
        with
        | ex -> Trace.TraceError($"Error in MeshVisualization: {ex.Message}")

[<Transaction(TransactionMode.Manual)>]
[<UsedImplicitly>]
type MeshesVisualization() =
    inherit ExternalCommand()

    override this.Execute() =
        try
            let meshRefs = this.Application.ActiveUIDocument.Selection.PickObjects(ObjectType.Element, "Select Mesh Elements")
            let meshes =
                meshRefs
                |> Seq.collect (fun mRef -> getMeshes (this.Application.ActiveUIDocument.Document.GetElement(mRef)))
                |> Seq.toList
            Trace.Write(meshes)
        with
        | ex -> Trace.TraceError($"Error in MeshesVisualization: {ex.Message}")

File: Samples/CSharpDemo/MeshVisualization.cs
using Autodesk.Revit.Attributes;
using Autodesk.Revit.UI.Selection;
using Nice3point.Revit.Toolkit.External;
using CSharpDemo.Extensions;
using System.Diagnostics;

namespace CSharpDemo;

[Transaction(TransactionMode.Manual)]
[UsedImplicitly]
public class MeshVisualization : ExternalCommand
{
    public override void Execute()
    {
        try
        {
            var meshRef = Application.ActiveUIDocument.Selection.PickObject(ObjectType.Element, "Select Mesh Element");
            var mesh = Application.ActiveUIDocument.Document.GetElement(meshRef).GetMeshes();
            Trace.Write(mesh.First());
        }
        catch (Exception e)
        {
            Trace.TraceError($"Error in MeshVisualization: {e.Message}");
        }
    }
}

[Transaction(TransactionMode.Manual)]
[UsedImplicitly]
public class MeshesVisualization : ExternalCommand
{
    public override void Execute()
    {
        try
        {
            var meshRefs = Application.ActiveUIDocument.Selection.PickObjects(ObjectType.Element, "Select Mesh Elements");
            var meshes = meshRefs.SelectMany(mRef => Application.ActiveUIDocument.Document.GetElement(mRef).GetMeshes()).ToList();
            Trace.Write(meshes);
        }
        catch (Exception e)
        {
            Trace.TraceError($"Error in MeshesVisualization: {e.Message}");
        }
    }
}